code/modules/mob/living/carbon/life.dm

```diff
@@ -245,6 +275,33 @@
 			if(!.)
 				return FALSE //to differentiate between no internals and active, but empty internals
 
+// Make corpses rot, emitting miasma
+/mob/living/carbon/proc/rot()

```

Comment:
Maybe also check if the bodytemperature is very low, and that the mob_biotype is MOB_ORGANIC or MOB_UNDEAD since i doubt that golems would rot

Reply:
Done